### PR TITLE
Remove the legacy networking stack.

### PR DESCRIPTION
**Pull request by Ms2ger**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14360)

### CODE REVIEW

File: components/net/http_loader.rs [hunk 1]
```diff
@@ -535,33 +397,6 @@ pub fn set_cookies_from_headers(url: &ServoUrl, headers: &Headers, cookie_jar: &
     }
 }
 
-fn update_sts_list_from_response(url: &ServoUrl, response: &HttpResponse, hsts_list: &Arc<RwLock<HstsList>>) {
```
Comment (jdm) [inline components/net/http_loader.rs]: Let's at least file an issue about reimplementing STS for fetch.

File: components/net/http_loader.rs [hunk 2]
```diff
@@ -662,63 +497,6 @@ pub fn send_response_to_devtools(devtools_chan: &Sender<DevtoolsControlMsg>,
...
-                                               referrer_policy.clone(),
-                                               referrer_url.clone(),
-                                               url.clone());
-
-    if let Some(referrer_val) = referrer_url.clone() {
-        headers.set(Referer(referrer_val.into_string()));
-    }
-}
-
-fn set_auth_header(headers: &mut Headers,
```
Comment (jdm) [inline components/net/http_loader.rs]: Let's file an issue about reimplementing HTTP auth for fetch.

Comment (Ms2ger) [inline components/net/http_loader.rs]: https://github.com/servo/servo/issues/14305

File: components/net/http_loader.rs (hunk 1 above)
Comment (Ms2ger) [inline components/net/http_loader.rs]: https://github.com/servo/servo/issues/14363

**Pull request merged**